immuneML/encodings/pdbencoding/PDBEncoder.py

```diff
@@ -0,0 +1,250 @@
+import matplotlib.colors
+import numpy
+from Bio.PDB import PDBParser
+from immuneML.data_model.encoded_data.EncodedData import EncodedData
+
+from immuneML.data_model.dataset.PDBDataset import PDBDataset
+from immuneML.encodings.DatasetEncoder import DatasetEncoder
+from immuneML.encodings.EncoderParams import EncoderParams
+import numpy as np
+from immuneML.data_model.receptor.RegionType import RegionType
+
+
+class PDBEncoder(DatasetEncoder):
+
+
+
+    @staticmethod
+    def build_object(dataset=None, **params):
+        return PDBEncoder(**params)
+
+    def __init__(self, name: str = None, region_type: RegionType = None):
+
+        self.name = name
+        self.region_type = region_type
+
+    def encode(self, dataset, params: EncoderParams):
+
+        structures =[]
+
+        pdbParser = PDBParser(
```

Comment:
parsing the PDB seems like it should be a responsibility of the Dataset. There may be many different encodings, so anything that is specific to this dataset that you would often want to re-use in other encodings or reports would ideally be a property of the dataset. 

I think it would be cleaner if PDBDataset has a function get_files that yields the files (like get_data does currently), and that get_data yields the PDB classes/parsed data. 

one thing we currently don't have for the PDBDataset, and do have for the other datasets, is an implementation of a 'single object':
- a RepertoireDataset consists of Repertoire objects
- a SequenceDataset consists of ReceptorSequence objects
- a ReceptorDataset consists of Receptor objects

for PDBDataset, we don't have PDBStructure objects. Perhaps this is not necessary, and perhaps the models for the ParserObject themselves are sufficient as 'single objects'. But we should keep this in mind and consider if it is beneficial to implement a PDBStructure object (in which case PDBDataset stores self.structures, a list of PDBStructure objects, and get_data returns self.structures). In this case, additional useful information can be stored inside PDBStructure upon import. Including for example:
- metadata like receptor or chain types (TCR, alpha/beta)
- boolean: whether the structure also contains an antigen or not
- maybe: amino acid translations of the CDR3 for easy access? v/j genes if they can easily be retrieved?
- identifier
-  (maybe RegionType?? start/stop indices for the CDR3 if the structure is non-IMGT -> these would then be read from a metadata file)
- other kinds of per-structure metadata

some of this extra information may seem irrelevant now, but we have to consider that if this extension will be part of immuneML, it should be easily usable for many different purposes. The Encoder should do as little data parsing as possible. The responsibility of the encoder is to create some kind of data representation that is a (multidimensional) matrix. Any other functionality breaks the single responsibility principle.